### ISSUE TITLE
Matcher ScalarHasValue has to override describeMismatchSafely for better output

### ISSUE THREAD

**Issue by proshin-roman** [llorllale/cactoos-matchers#7]
This ticket is moved from https://github.com/yegor256/cactoos/issues/816

Currently, `ScalarHasValue` doesn't override method `describeMismatchSafely` that leads to the next problem:
```java
final ScalarHasValue<String> matcher =
    new ScalarHasValue<>(new IsEqual<>("expected"));
final Scalar<String> scalar = new UncheckedScalar<>(() -> "actual");
final StringDescription description = new StringDescription();
matcher.describeMismatch(scalar, description);
System.out.println(description.toString());
// the out is:
// was <org.cactoos.scalar.UncheckedScalar@368239c8>
// how it should look like:
// was "actual"
```
so it prints an actual version incorrectly: default toString method is called for the Scalar.

**Expected:**
- the value of scalar is printed as an actual value of matching.

**What should be fixed**
- class `ScalarHasValue` has to override method `describeMismatchSafely` and convert Scalar to String properly.

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/llorllale/cactoos-matchers/issues/7#issuecomment-388584791))

@llorllale Job [#7](https://github.com/llorllale/cactoos-matchers/issues/7) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @proshin-roman[/z](https://www.0crat.com/u/proshin-roman)

**Comment by 0crat** [comment]
The job [#7](https://github.com/llorllale/cactoos-matchers/issues/7) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/1361361625); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by victornoel** [comment]
@llorllale should other matchers be improved as well?

**Comment by llorllale** [comment]
@victornoel sure. You can also leave puzzles for continuing work past budget

**Comment by victornoel** [comment]
@0crat waiting for #14

**Comment by 0crat** [comment]
> @0crat waiting for #14  ([here](https://github.com/llorllale/cactoos-matchers/issues/7#issuecomment-390419612))

@victornoel The impediment for [#7](https://github.com/llorllale/cactoos-matchers/issues/7) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by victornoel** [comment]
@proshin-roman the MR #14 has been merged, this issue can now be closed :)

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#7](https://github.com/llorllale/cactoos-matchers/issues/7) is now out of scope

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Comment by 0pdd** [comment]
@proshin-roman the only puzzle [#20](https://github.com/llorllale/cactoos-matchers/issues/20) is solved here.

**Comment by 0pdd** [comment]
@proshin-roman the puzzle [#20](https://github.com/llorllale/cactoos-matchers/issues/20) is still not solved.

**Comment by 0pdd** [comment]
@proshin-roman the puzzle [#230](https://github.com/llorllale/cactoos-matchers/issues/230) is still not solved; solved: [#20](https://github.com/llorllale/cactoos-matchers/issues/20).
